packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_time_visualizer.dart

```diff
@@ -0,0 +1,325 @@
+          icon: Icons.grid_on,
+        ),
+      ];
+    }
+    final rasterPhase = frameAnalysis.rasterPhase;
+    return [
+      _FramePhaseBlockData(
+        title: rasterPhase.title,
+        duration: rasterPhase.duration,
+        flex: frameAnalysis.rasterFlex!,
+        icon: Icons.grid_on,
+      ),
+    ];
+  }
+}
+
+class _FrameBlockGroup extends StatelessWidget {
+  const _FrameBlockGroup({
+    Key? key,
+    required this.title,
+    required this.data,
+    required this.hasData,
+  }) : super(key: key);
+
+  final String title;
+
+  final List<_FramePhaseBlockData> data;
+
+  final bool hasData;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget content;
+    if (hasData) {
+      final totalFlex =
+          data.fold<int>(0, (current, block) => current + block.flex);
+      content = LayoutBuilder(
+        builder: (context, constraints) {
+          final adjustedBlockWidths =
+              adjustedWidthsForBlocks(constraints, totalFlex);
+          return Row(
+            children: [
+              for (var i = 0; i < data.length; i++)
+                _FramePhaseBlock(
+                  blockData: data[i],
+                  width: adjustedBlockWidths[i],
+                ),
+            ],
+          );
+        },
+      );
+    } else {
+      content = const Text('Data not available.');
+    }
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(title),
+        const SizedBox(height: denseSpacing),
+        content,
+      ],
+    );
+  }
+
+  /// Returns a list of adjusted widths for each block.
+  ///
+  /// The adjusted widths will ensure each block is at least
+  /// [_FramePhaseBlock.minBlockWidth] wide, and will modify surrounding block
+  /// widths to accomodate. f
+  List<double> adjustedWidthsForBlocks(
+    BoxConstraints constraints,
+    int totalFlex,
+  ) {
+    final unadjustedBlockWidths = data
+        .map(
+          (blockData) => constraints.maxWidth * blockData.flex / totalFlex,
+        )
+        .toList();
+
+    var adjustment = 0.0;
+    var widestBlockIndex = 0;
+    for (var i = 0; i < unadjustedBlockWidths.length; i++) {
+      final unadjustedWidth = unadjustedBlockWidths[i];
+      final currentWidestBlock = unadjustedBlockWidths[widestBlockIndex];
+      if (unadjustedWidth > currentWidestBlock) {
+        widestBlockIndex = i;
+      }
+      if (unadjustedWidth < _FramePhaseBlock.minBlockWidth) {
+        adjustment += _FramePhaseBlock.minBlockWidth - unadjustedWidth;
+      }
+    }
+
+    final adjustedBlockWidths = <double>[];
+    for (var i = 0; i < unadjustedBlockWidths.length; i++) {
+      final blockWidth = unadjustedBlockWidths[i];
+      var adjustedWidth = blockWidth;
+      if (i == widestBlockIndex) {
+        adjustedWidth -= adjustment;
+      }
+      adjustedWidth = math.max(adjustedWidth, _FramePhaseBlock.minBlockWidth);
+      adjustedBlockWidths.add(adjustedWidth);
+    }
```

Comment:
```
final adjustedBlockWidths = unadjustedBlockWidths.map(b => math.max(b,  _FramePhaseBlock.minBlockWidth));
adjustedBlockWidths[widestBlockIndex] -= adjustment; 
```
I think you can simplify this with a map and an index modification :)

Reply:
subtracting the adjustment needs to happen before we perform the `max` calculation so that we do not end up with an `adjustedWidth < _FramePhaseBlock.minBlockWidth` for the widest block index. 

I can simplify this though with another call to `max` when assigning `adjustedBlockWidths[widestBlockIndex]`.